lora-phy/src/sx127x/mod.rs

```diff
@@ -553,6 +543,29 @@ where
         // If no specific IRQ condition is met, return None
         Ok(None)
     }
+
+    async fn clear_irq_status(&mut self) -> Result<(), RadioError> {
+        self.write_register(Register::RegIrqFlags, 0xffu8).await // clear all interrupts
+    }
+
+    /// Process the radio IRQ. Log unexpected interrupts. Packets from other
+    /// devices can cause unexpected interrupts.
+    ///
+    /// NB! Do not await this future in a select branch as interrupting it
+    /// mid-flow could cause radio lock up.
+    async fn process_irq_event(
+        &mut self,
+        radio_mode: RadioMode,
+        cad_activity_detected: Option<&mut bool>,
+        clear_interrupts: bool,
+    ) -> Result<Option<IrqState>, RadioError> {
+        let irq_state = self.get_irq_state(radio_mode, cad_activity_detected).await;
+        if clear_interrupts && irq_state.as_ref().is_ok_and(|state| state.is_some()) {
```

Comment:
Before we cleared the irq flags if there were any unconditionally. If irq_state is an error ex. `Err(RadioError::TransmitTimeout)` would clear_irq_status trigger?.